### ISSUE TITLE
evdev type mismatch

### ISSUE THREAD

**Issue by glyh** [canselcik/libremarkable#106]
I've got this error message for evdev, when I was running `make TARGET=armv7-unknown-linux-gnueabihf deploy-x-demo`:
```
error[E0308]: mismatched types
   --> /cargo/registry/src/github.com-1ecc6299db9ec823/evdev-0.11.7/src/lib.rs:390:29
    |
390 |     let dur = Duration::new(tv.tv_sec.unsigned_abs(), tv.tv_usec as u32 * 1000);
    |               ------------- ^^^^^^^^^^^^^^^^^^^^^^^^ expected `u64`, found `u32`
    |               |
    |               arguments to this function are incorrect
    |
note: associated function defined here
   --> /rust/lib/rustlib/src/rust/library/core/src/time.rs:185:18
    |
185 |     pub const fn new(secs: u64, nanos: u32) -> Duration {
    |                  ^^^
help: you can convert a `u32` to a `u64`
    |
390 |     let dur = Duration::new(tv.tv_sec.unsigned_abs().into(), tv.tv_usec as u32 * 1000);
    |                                                     +++++++

For more information about this error, try `rustc --explain E0308`.
error: could not compile `evdev` due to previous error
```

**Comment by LinusCDE** [comment]
Hi.

At first I couldn't reproduce this error. But then noticed that your version of evdev was newer then the one in my (older) `Cargo.lock` file.

Seems something between evdev 0.11.3 and 0.11.7 changed that broke it. I'm looking into it and will create a PR to fix it shortly.

Good catch! :+1:

**Comment by LinusCDE** [comment]
The same compilation error was found 9 days ago in issue https://github.com/emberian/evdev/issues/95 (which lead to the PR fixing it)

**Comment by LinusCDE** [comment]
The issue at hand is fixed in the [0.6.1 release](https://crates.io/crates/libremarkable/0.6.1).

As said, when the new evdev release is around, we can update to it and close this issue. That will then be included in a future `0.6.2` or  `0.7.0` release when we get to do that (not making that release for this fix alone though).
